<a href="https://colab.research.google.com/github/dhiyanajasdeen345-wq/data-analytics-final-project/blob/main/Notebook_1_Data_loading_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#📊 E-Commerce Sales Data Analysis


---

##1. Introduction

###1.1 Purpose of the Analysis


The purpose of this analysis is to explore and understand the structure and composition of the e-commerce sales dataset. The dataset contains transaction-level information that can be used to analyze customer behavior, product performance, and sales patterns across different regions and categories.


This initial phase focuses on gaining a clear understanding of the dataset by examining its structure, features, and overall organization before proceeding to deeper analytical steps.

---



###2. Data Source

The dataset used in this analysis is an e-commerce sales dataset that represents transactions typically generated in an online retail environment. It is designed to simulate real-world business scenarios and includes information related to customers, products, and financial metrics.

Such datasets are commonly derived from internal sales systems, order management platforms, or generated for analytical and educational purposes. The structure and variables closely resemble those found in real e-commerce platforms, making it suitable for exploratory data analysis and business insights.

---



###3. Data Loading

3.1 Importing Required Libraries


In [10]:
import pandas as pd
import numpy as np

These libraries are essential for handling structured data and performing numerical operations efficiently.

---

###3.2 Loading the Dataset

In [11]:
df = pd.read_excel("Sales_ecommerce_data set (1).xlsx")

The dataset is loaded into a pandas DataFrame, which provides a flexible and efficient structure for data manipulation and analysis.

---

##4. Initial Data Inspection

###4.1 Viewing the First Rows


In [12]:
df.head()

,Order_ID,Customer_Name,Region,City,Product_Category,Sales_Amount,Profit,Discount_Percent,Quantity,Payment_Method,Order_Date,Customer_Rating
0,ORD10000,Rachel Booth,North,Hyderabad,NaN,6241.79,1297.37,5.0,4,UPI,16-10-2024,1.0
1,ORD10001,Chase Johnston,South,Hyderabad,NaN,4654.34,-335.28,20.0,1,Cash,31-01-2023,3.0
2,ORD10002,Johnathan Castillo,south,Bangalore,Home,6619.22,1438.65,5.0,7,Net Banking,08-04-2025,3.0
3,ORD10003,Pamela Garcia,West,Kochi,Sports,8807.57,-1551.03,15.0,9,upi,10-09-2024,3.0
4,ORD10004,Sara Simmons,North,Bangalore,Electronics,4414.62,-714.04,NaN,4,Net Banking,14-08-2024,3.0


The first few rows of the dataset provide an initial understanding of how the data is organized. Each row corresponds to a unique transaction, and the columns represent different attributes such as customer information, product category, and financial details.


---

###4.2 Viewing the Last Rows


In [13]:
df.tail()

,Order_ID,Customer_Name,Region,City,Product_Category,Sales_Amount,Profit,Discount_Percent,Quantity,Payment_Method,Order_Date,Customer_Rating
3045,ORD11547,Anna Miller,south,Hyderabad,NaN,5754.48,-365.22,5.0,6,upi,06-05-2024,3.0
3046,ORD12425,Virginia Delgado,South,Hyderabad,Beauty,6285.64,1640.97,0.0,9,Net Banking,07-05-2024,1.0
3047,ORD10014,Keith Kirk,West,Kochi,Fashion,687.71,116.52,10.0,8,Card,2023/11/16,5.0
3048,ORD12567,Aaron Nicholson,South,Chennai,Home,6627.84,320.69,15.0,4,UPI,16-11-2023,3.0
3049,ORD11600,Nancy Myers,North,Hyderabad,NaN,4542.76,-320.24,20.0,2,UPI,05-04-2024,2.0


Inspecting the last rows helps confirm that the dataset maintains a consistent structure throughout. There are no visible formatting inconsistencies or structural deviations at the end of the dataset.



---


###4.3 Dataset Shape

In [14]:
df.shape

(3050, 12)

The dataset contains 3050 rows and 12 columns, indicating that there are 3050 individual transactions and 12 distinct features describing each transaction.

This size is sufficient for performing meaningful exploratory data analysis and identifying patterns within the data.

---

###4.4 Column Names


In [15]:
df.columns

Index(['Order_ID', 'Customer_Name', 'Region', 'City', 'Product_Category',
       'Sales_Amount', 'Profit', 'Discount_Percent', 'Quantity',
       'Payment_Method', 'Order_Date', 'Customer_Rating'],
      dtype='object')

The dataset includes the following columns :

### Feature Explanation


####Order_ID:
Unique identifier assigned to each transaction to distinguish records.


####Customer_Name:
Name of the customer who placed the order.


####Region:
Geographical region where the transaction occurred.


####City:
City associated with the customer or order.


####Product_Category:
Category to which the purchased product belongs.


####Sales_Amount:
Total revenue generated from the transaction.


####Profit:
Net profit obtained after deducting costs.


####Discount_Percent:
Percentage of discount applied to the product.


####Quantity:
Number of units purchased in the transaction.


####Payment_Method:
Mode of payment used by the customer.


####Order_Date:
Date on which the transaction took place.


####Customer_Rating:
Rating provided by the customer based on their experience.

The columns collectively describes different dimensions of each transactions , including identification, customer details, product classification, and financial performance.



---

##5. Observations

###5.1 Structural Observations

The dataset is well-structured in a tabular format with clearly defined rows and columns. Each row represents a single transaction, and there is no indication of nested or unstructured data. This makes the dataset suitable for direct analysis without requiring extensive restructuring.

---

###5.2 Consistency and Formatting

From the initial inspection using "head()" and "tail()", the dataset appears consistent across all entries. The columns are properly aligned, and there are no visible anomalies such as shifted values or irregular formatting.

---

###5.3 Feature Composition

The dataset includes a diverse set of features:

- Categorical Features: Region, City, Product_Category, Payment_Method
- Numerical Features: Sales_Amount, Profit, Quantity, Discount_Percent
- Date Feature: Order_Date
- Identifier: Order_ID

This combination allows multiple analytical perspectives, including categorical comparison, numerical analysis, and time-based trends.

---

###5.4 Transaction-Level Detail

Each record captures detailed information about a transaction, including who made the purchase, what was purchased, how much was spent, and how the payment was made. This level of granularity enables deeper insights into customer behavior and sales performance.

---

###5.5 Readiness for Analysis

Based on the initial inspection, the dataset appears clean and structured, making it ready for further preprocessing steps such as checking missing values, data types, and performing exploratory data analysis.
